Add DWPP Description

decwest opened the pull request:
<!-- Please fill out the following pull request template for non-trivial changes to help us process your PR faster and more efficiently.-->

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | https://github.com/ros-navigation/navigation2/pull/5591 |
| Does this PR contain AI-generated software? | No |
---

## Description of contribution in a few bullet points

- Updated the documentation to reflect implenmentation of DWPP in navigation2.
- Added descriptions of the parameters introduced for the DWPP implementation.

<!--
* Updated the documentation to reflect corresponding changes in navigation2.
* Reorganised some sections for better clarity.
-->

Decwest commented on configuration/packages/configuring-regulated-pp.rst:
```diff
@@ -12,6 +12,7 @@ It regulates the linear velocities by curvature of the path to help reduce overs
 It also better follows paths than any other variation currently available of Pure Pursuit.
 It also has heuristics to slow in proximity to other obstacles so that you can slow the robot automatically when nearby potential collisions.
 It also implements the Adaptive lookahead point features to be scaled by velocities to enable more stable behavior in a larger range of translational speeds.
+It also considers the robot’s velocity and acceleration constraints during velocity command computation using `Dynamic Window Pure Pursuit <https://github.com/Decwest/nav2_dynamic_window_pure_pursuit_controller/blob/main/algorithm.md>`_ algorithm.
```
Since the paper has not been published yet, I have included a link to the webpage describing the algorithm. The paper will be published within a few months, and once it becomes available, I will update the link accordingly.

SteveMacenski commented on configuration/packages/configuring-regulated-pp.rst:
```diff
@@ -22,7 +23,7 @@ If you use the Regulated Pure Pursuit Controller algorithm or software from this
 Regulated Pure Pursuit Parameters
 *********************************
 
-:desired_linear_vel:
```
In the migration guide, we need to mention this change. I'd also put in the description a note that it is `desired_linear_vel` in Kilted and older.

Decwest replied:
I see, should I write a migration guide like this?
https://docs.nav2.org/migration/Jazzy.html#regulatedpurepursuit-controller-rpp-new-parameter-stateful

Decwest replied:
Also, which migration guide should I update? For example, if this DWPP PR ends up being backported to both Kilted and Jazzy, should I add the notes to something like the “Iron to Jazzy” guide?

SteveMacenski commented on configuration/packages/configuring-regulated-pp.rst:
```diff
@@ -367,6 +434,18 @@ Regulated Pure Pursuit Parameters
+  ============== =============================
+
+  Description
+    Whether to use the Dynamic Window Pure Pursuit (DWPP) Algorithm. This algorithm computes optimal path tracking velocity commands under velocity and acceleration constraints.
```
Migth be good to put a link to the paper / title of the paper for folks to review

SteveMacenski commented:
@Decwest last updates here are the migration guide entry for this change to add the feature + parameter renaming. This is important for folks to be aware of changes they need to make to their configurations when upgrading distributions and expose that there's this new feature they may want to try out.

SteveMacenski commented on migration/Kilted.rst:
```diff
@@ -562,3 +562,114 @@ The UI workflow is now organized into two primary navigation modes:
+
+  Description
+    Whether to enable the Dynamic Window Pure Pursuit (DWPP) algorithm.
+
```
```suggestion
```

SteveMacenski commented on migration/Kilted.rst:
```diff
@@ -562,3 +562,114 @@ The UI workflow is now organized into two primary navigation modes:
+  ============== ===========================
+
+  Description
+    The maximum angular deceleration (rad/s^2) used when `use_dynamic_window` is `true`.
```
```suggestion
```

SteveMacenski commented on migration/Kilted.rst:
```diff
@@ -562,3 +562,114 @@ The UI workflow is now organized into two primary navigation modes:
+Add Dynamic Window Pure Pursuit Option to Regulated Pure Pursuit Controller
+---------------------------------------------------------------------------
+
+In `PR #5783 <https://github.com/ros-navigation/navigation2/pull/5783>`_, an option was added to enable the Dynamic Window Pure Pursuit (DWPP) algorithm in the Regulated Pure Pursuit controller. When this option is enabled, velocity and acceleration constraints are explicitly considered when computing command velocities.
```
```suggestion
In `PR #5783 <https://github.com/ros-navigation/navigation2/pull/5783>`_, an option was added to enable the Dynamic Window Pure Pursuit (DWPP) algorithm in the Regulated Pure Pursuit controller. When this option is enabled, velocity and acceleration constraints are explicitly considered when computing command velocities. See the Configuration Guide for the new parameters associated with this feature.
```

SteveMacenski reviewed (commented):
LGTM, I just reduced the migration guide bit to be more concise, this should be an overview not a detailed blow by blow

Please also fix the merge conflicts

Decwest commented:
@SteveMacenski @mini-1235 

Thank you for reviewing the documents.

I have fixed the merge conflicts and addressed the comments I received.
I would appreciate it if you could take a look when you have some time.

P.S. I am currently preparing a journal paper on DWPP, which includes real-robot experimental results comparing PP, APP, RPP, and DWPP using the Nav2 implementation :)
The submission deadline is January 21, so I am planning to upload a preprint to arXiv toward the end of this month and include the link in this document as well.

I would like to ask for your advice on one point.
Do you think it is realistic to have DWPP merged into the main branch by January 21?
If integration is possible, would it be acceptable to include a statement such as the following in the journal paper to highlight that DWPP has been integrated into Nav2?

> The proposed method is integrated into the official Nav2 repository and is publicly available.

Decwest commented:
@SteveMacenski 
Thank you for checking.
I added a concise explanation of the function and its value to the use_dynamic_window description entry.
https://github.com/ros-navigation/docs.nav2.org/blob/78081abfc8ce38289db4b57970adc29cb81a82e2/configuration/packages/configuring-regulated-pp.rst?plain=1#L424

Also, I will move my DWPP Nav2 plugin repository to a public archive after this is merged!

SteveMacenski commented:
Sounds good to me
